lib/experimental/media/class-gutenberg-rest-attachments-controller.php

```diff
@@ -0,0 +1,350 @@
+<?php
+/**
+ * Class Gutenberg_REST_Attachments_Controller.
+ *
+ * @package MediaExperiments
+ */
+
+/**
+ * Class Gutenberg_REST_Attachments_Controller.
+ */
+class Gutenberg_REST_Attachments_Controller extends WP_REST_Attachments_Controller {
+	/**
+	 * Registers the routes for attachments.
+	 *
+	 * @see register_rest_route()
+	 */
+	public function register_routes(): void {
+		parent::register_routes();
+
+		$args                       = $this->get_endpoint_args_for_item_schema();
+		$args['generate_sub_sizes'] = array(
+			'type'        => 'boolean',
+			'default'     => true,
+			'description' => __( 'Whether to generate image sub sizes.', 'gutenberg' ),
+		);
+		$args['convert_format']     = array(
+			'type'        => 'boolean',
+			'default'     => true,
+			'description' => __( 'Whether to support server-side format conversion.', 'gutenberg' ),
+		);
+
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_items' ),
+					'permission_callback' => array( $this, 'get_items_permissions_check' ),
+					'args'                => $this->get_collection_params(),
+				),
+				array(
+					'methods'             => WP_REST_Server::CREATABLE,
+					'callback'            => array( $this, 'create_item' ),
+					'permission_callback' => array( $this, 'create_item_permissions_check' ),
+					'args'                => $args,
+				),
```

Comment:
No, because `WP_REST_Attachments_Controller::register_routes()` doesn't use that, so need to register the routes  anew.

For a core merge it will look differently anyway, then it can just be added to the class directly. For temporary code like this it's totally fine.

Reply:
> No, because `WP_REST_Attachments_Controller::register_routes()` doesn't use that, so need to register the routes anew.

Not sure that's correct. As far as I can tell, it _does_ use it, the actual route is registered in the parent class `WP_REST_Posts_Controller` and it uses `get_endpoint_args_for_item_schema()`. So I don't see why we shouldn't use that here instead of duplicating the code to register.